core/app/models/spree/tax_category.rb

```diff
@@ -4,7 +4,10 @@ class TaxCategory < Spree::Base
     validates :name, presence: true
     validates_uniqueness_of :name, unless: :deleted_at
 
-    has_many :tax_rates, dependent: :destroy, inverse_of: :tax_category
+
+    has_many :tax_rate_categories, class_name: Spree::TaxRateCategory, dependent: :delete_all
```

Comment:
I wonder if we should even really do `dependent: ` at all here?  With our new data model it might be a good thing that destroying a tax category will fail (due to FK constraints) if it's in use by at least one tax rate?